controllers/urls.js

```diff
@@ -0,0 +1,131 @@
+const { Url, User } = require("../models/schema");
+const shuffle = require("../lib/utils").shuffle;
+const TagControl = require("./tags");
+const UserControl = require("./users");
+
+const createURL = async (req, res) => {
+  try {
+    const [title, url, tags] = req.body.params;
+    const submittedBy = req.body.userId;
+    const existingUrl = await Url.findOne({ url });
+    if (existingUrl) {
+      return res.status(400).json({ error: "URL already exists" });
+    }
+
+    const newUrl = await Url.create({ title, url, submittedBy, tags });
+
+    TagControl.attachURL(tags, newUrl.id);
+    UserControl.attachURL(submittedBy, newUrl.id);
+
+    return res.status(201).json(newUrl);
+  } catch (err) {
+    console.error(err);
+    return res.status(500).json({ error: "Server error" });
+  }
+};
+
+const deleteURL = async (req, res) => {
+  try {
+    const { urlId } = req.params;
+    const url = await Url.findById(urlId);
+    if (!url) {
+      return res.status(404).json({ error: "URL not found" });
+    }
+
+    await TagControl.detachURL(url.tags, urlId);
+    await UserControl.detachURL(url.submittedBy, urlId);
+
+    await Url.deleteOne({ _id: urlId });
+
+    return res.status(200).json({ message: "URL removed successfully" });
+  } catch (err) {
+    console.error(err);
+    return res.status(500).json({ error: "Server error" });
+  }
+};
+
+const __getURLsFromDb = async (tags, page = 1, limit = 100) => {
+  if (!tags) tags = "all";
+  try {
+    const skipCount = (page - 1) * limit;
+
+    const query = tags.includes("all")
+      ? Url.find()
+      : Url.find({ tags: { $in: tags } });
+
+    const results = await query
+      .skip(skipCount)
+      .limit(limit)
+      .populate("tags", "name");
+
+    return results;
+  } catch (error) {
+    return { error };
+  }
+};
+
+const getMixedURLs = async (req, res) => {
+  try {
+    const { tags, page, limit } = req.query;
+
+    const [urls, count] = await Promise.all([
+      __getURLsFromDb(tags, page, limit),
+      tags.includes("all")
```

Comment:
tags will never get "all" . In __getURLsFromDb we pass tags, page and limit but it won't change the tag value in getMixedURLs function.